src/index.ts

```diff
@@ -138,6 +140,66 @@ export const createInstance = (config: Config): Instance => {
 
       return tx;
     },
+
+    buyNow: async (
+      signer: ethers.Signer,
+      params: BuyNowParams
+    ): Promise<ethers.ContractTransaction> => {
+      const zAuction = await getZAuctionContract(
+        signer,
+        config.zAuctionAddress
+      );
+
+      const nftContract = await getERC721Contract(
+        config.web3Provider,
+        config.tokenContract
+      );
+
+      const isApproved = await actions.isZAuctionApprovedNftTransfer(
+        await signer.getAddress(),
+        config.zAuctionAddress,
+        nftContract
+      );
+
+      if (!isApproved) throw Error("User is not approved for NFT Transfer");
+
```

Comment:
An additional check needs to happen here: 

When a user buys an NFT, zAuction needs to be able to transfer their WILD tokens.
You need to verify that the `buyer` has approved zAuction to transfer WILD tokens on their behalf.

You can get the quantity of WILD tokens zAuction is allowed to transfer on behalf of a user via `actions.getZAuctionSpendAllowance()`

Thus the check
 `getZAuctionSpendAllowance() >= params.amount`.
Will ensure zAuction can transfer enough tokens.